### PR TITLE
Rewritten structuring.md

### PR DESCRIPTION
**Pull request by jlitewski**
Rewrote most of the article to flow better, and to show better and more robust examples.

### CODE REVIEW

File: docs/gettingstarted/structuring.md [hunk 1]
```diff
@@ -1,54 +1,83 @@
...
-After the top level package (if you have one) you append a unique name for your mod, such as `examplemod`. In our case it will end up as `com.example.examplemod`.
+After you pick a top level package, you should append a unique name to it for your mod. Some good examples would be `com.examplemod.awesomesauce` or `username.mods.awesomesauce`, where **AwesomeSauce** is the name of our mod.
 
 The Mod File
 ------------
 
-Generally, we'll start with a file named after your mod, and put into your package. This is the *entry point* to your mod
-and will contain some special indicators marking it as such.
+Generally, we'll start with a file named after your mod, and put into your package. This is the *entry point* to your mod and will contain some special annotations marking it as such. These annotations are `@Mod`, `@Instance`, and `@EventHandler`. Let's take a look at this example:
+
```
Comment (bonii-xx) [inline docs/gettingstarted/structuring.md]: The example code should be in a codeblock

File: docs/gettingstarted/structuring.md [hunk 2]
```diff
@@ -1,54 +1,83 @@
...
+            // some example code
+            System.out.println("DIRT BLOCK >> "+Blocks.DIRT.getUnlocalizedName());
+        }
+    }
 
 What is `@Mod`?
 -------------
 
-This is an annotation indicating to the Forge Mod Loader that the class is a Mod entry point. It contains various metadata about the mod. It also designates the class that will receive `@EventHandler` events. More information can be found at... (Coming Soon)
+`@Mod` is the annotation indicating that the class is a *Mod entry point* for Forge Mod Loader. It contains various bits of Metadata that the mod will use as well as modifiers that tell Forge how to handle your mod.
```
Comment (bonii-xx) [inline docs/gettingstarted/structuring.md]: What modifiers are you talking about here?

File: docs/gettingstarted/structuring.md [hunk 3]
```diff
@@ -1,54 +1,83 @@
...
 
-A common subpackage strategy has packages for `common` and `client` code, which is code that can be run on server/client and client, respectively. Inside the `common` package would go things like Items, Blocks, and Tile Entities (which can each in turn be another subpackage). Things like GUIs and Renderers would go inside the `client` package.
+What is `@EventHandler`?
+-----------------------
+`@EventHandler` is an annotation used to mark whatever method it's attached to as a handler for a *Forge Lifecycle Event*. There are two main categories of Lifecycle Events: **Standard** and **Server**. **Standard Lifecycle Events** are always fired when the client and server first starts up, and **Server Lifecycle Events** are only fired when the server, either internal or dedicated, starts up. There are some specialized events as well, but those will be covered in a different atricle.
 
-!!! note
+For a simple mod, *FMLPreInitializationEvent*, *FMLInitializationEvent*, and *FMLPostInitializationEvent* should be more than enough to set up everything you need.
```
Comment (bonii-xx) [inline docs/gettingstarted/structuring.md]: Add a link to the [loading stages](https://mcforge.readthedocs.io/en/latest/conventions/loadstages/) here.

**Comment by bonii-xx** [comment]
I like it. Besides the few notes above.

File: docs/gettingstarted/structuring.md [hunk 4]
```diff
@@ -1,54 +1,83 @@
...
+    import net.minecraft.init.Blocks;
+    import net.minecraftforge.fml.common.Mod;
+    import net.minecraftforge.fml.common.Mod.EventHandler;
+    import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+        
+    @Mod(modid = "awesomesauce", name = "AwesomeSauce", version = "0.1")
+    public class AwesomeSauceMod
+    {
+        @Instance
+        public static AwesomeSauceMod instance = new AwesomeSauceMod();
```
Comment (diesieben07) [inline docs/gettingstarted/structuring.md]: No.

File: docs/gettingstarted/structuring.md [hunk 5]
```diff
@@ -1,54 +1,83 @@
...
+* `version` is the version of the mod, and takes a `String`. There is no set standard for how you version things, but it's recommended that you read the [versioning](https://mcforge.readthedocs.io/en/latest/conventions/versioning/) article.
 
-Keeping Your Code Clean Using Sub-packages
-------------------------------------------
+Simple, right? `@Mod` has a wide range of other things you can throw at it, but those will be covered in the articles that use them.
 
-Rather than clutter up a single class and package with everything, it is recommended you break your mod into subpackages.
+What is `@Instance`?
+-------------------
+`@Instance` is actually a child annotation of `@Mod`. It basically just gives Forge and other Mod Developers a way to instance your mod if they need to.
```
Comment (diesieben07) [inline docs/gettingstarted/structuring.md]: This is a horrible and inaccurate description of what @Instance does.

File: docs/gettingstarted/structuring.md [hunk 6]
```diff
@@ -1,54 +1,83 @@
...
 
 What is `@Mod`?
 -------------
 
-This is an annotation indicating to the Forge Mod Loader that the class is a Mod entry point. It contains various metadata about the mod. It also designates the class that will receive `@EventHandler` events. More information can be found at... (Coming Soon)
+`@Mod` is the annotation indicating that the class is a *Mod entry point* for Forge Mod Loader. It contains various bits of Metadata that the mod will use as well as modifiers that tell Forge how to handle your mod.
 
-You can find an example mod in the [Forge src download](http://files.minecraftforge.net/).
+Now, the `@Mod` annotation can take quite a few arguments, but the only three *required* ones are `modid`, `name`, and `version`.
+* `modid` is the unique identifier and *internal* name of the mod, and takes a `String`. Usually, this is the same name as the mod, but in all lowercase. It's used internally by Forge and by other Mod Developers that wish to find your mod through Forge (for instance, to use your APIs).
```
Comment (shadowfacts) [inline docs/gettingstarted/structuring.md]: You should note the new requirements here. (It must be all lowercase and not more than 64 characters long)

File: docs/gettingstarted/structuring.md (hunk 1 above)
Comment (HACKhalo2) [inline docs/gettingstarted/structuring.md]: It should already be in a code block.

`This is a line of code that uses a tilde`

    This is a block of code
    that uses 4 spaces per line
        to show indentation and stuff

Unless there is a different one I don't know about?

File: docs/gettingstarted/structuring.md (hunk 3 above)
Comment (HACKhalo2) [inline docs/gettingstarted/structuring.md]: Good Idea

File: docs/gettingstarted/structuring.md [hunk 7]
```diff
@@ -1,54 +1,83 @@
...
+        @Instance
+        public static AwesomeSauceMod instance = new AwesomeSauceMod();
+        
+        @EventHandler
+        public void init(FMLInitializationEvent event)
+        {
+            // some example code
+            System.out.println("DIRT BLOCK >> "+Blocks.DIRT.getUnlocalizedName());
+        }
+    }
```
Comment (HACKhalo2) [inline docs/gettingstarted/structuring.md]: I mean, yea, I did take it from the MDK, but having at bare bones example in documentation for people that don't have the MDK configured yet and they are looking at the documentation before diving in should at least be considered. I also find it hard to follow documentation if you have to follow a link back and forth from code and the docs. I'll add a link to the example mod on the github and remove the code block if nessasary

File: docs/gettingstarted/structuring.md (hunk 6 above)
Comment (HACKhalo2) [inline docs/gettingstarted/structuring.md]: I actually didn't know this. Noted.

File: docs/gettingstarted/structuring.md (hunk 5 above)
Comment (HACKhalo2) [inline docs/gettingstarted/structuring.md]: From the JavaDocs:

    /**
      * Populate the annotated field with the mod instance based on the specified ModId. This can be used
      * to retrieve instances of other mods.
      * @author cpw
    **/

That's what I based it on, however I'm also not entirely sure what it does anyways, so I'll look around and learn about it and rewrite this section after I educate myself.

File: docs/gettingstarted/structuring.md (hunk 2 above)
Comment (HACKhalo2) [inline docs/gettingstarted/structuring.md]: Yea, that was kinda sparce, huh? I'll flesh it out a bit.

File: docs/gettingstarted/structuring.md (hunk 7 above)
Comment (diesieben07) [inline docs/gettingstarted/structuring.md]: What I am saying is that this code example does not provide any additional value besides what you have already explained.

File: docs/gettingstarted/structuring.md (hunk 5 above)
Comment (diesieben07) [inline docs/gettingstarted/structuring.md]: It inserts the instance of the mod specified via modID in the annotation into the field. You don't populate the field yourself for a start (like you did above).

Comment (HACKhalo2) [inline docs/gettingstarted/structuring.md]: Oh, huh. I could of swore that is how you use to do it. Maybe an artifact I picked up from older versions of Forge? I've been tinkering with it since Minecraft 1.0, so it's been a log time

File: docs/gettingstarted/structuring.md (hunk 7 above)
Comment (HACKhalo2) [inline docs/gettingstarted/structuring.md]: Fair enough

File: docs/gettingstarted/structuring.md (hunk 5 above)
Comment (diesieben07) [inline docs/gettingstarted/structuring.md]: Many people do it, but it's useless and misleading, as Forge will immediately overwrite the field anyways.

File: docs/gettingstarted/structuring.md (hunk 7 above)
Comment (LexManos) [inline docs/gettingstarted/structuring.md]: On top of that, copy pasta, even if it is from our example mod, is bad.

File: docs/gettingstarted/structuring.md [hunk 8]
```diff
@@ -1,54 +1,79 @@
...
 ---------
 
-Pick a unique package name. If you own a URL associated with your project, you can use it as your top level package. For example if you own "example.com", you may use `com.example` as your top level package.
+Your mod should have a unique package name to avoid conflicts with anything else that might be loaded in from a Classloader. If you own a domain you would like to associate or have a domain associated with other projects already, you can use it as your **top level package**. For example, if you own the domain *"examplemod.com"*, you can choose to use `com.examplemod` as your top level package.
 
-!!! important
+**HOWEVER**
 
-    If you do not own a domain, do not use it for your top level package. It is perfectly acceptable to start your package with anything, such as your name/nickname, or the name of the mod.
+If you do not own a domain, **do *NOT* use it for your top level package!**
```
Comment (LexManos) [inline docs/gettingstarted/structuring.md]: The second preferred option is to use your email, so people have a way of associating you with something you own.
EXA: com.gmail.lexmanos

File: docs/gettingstarted/structuring.md [hunk 9]
```diff
@@ -1,54 +1,79 @@
...
-A common class naming scheme allows easier deciphering of what a class is, and also makes it easier for someone developing with your mod to find things.
-
-For Example:
-
-* An `Item` called `PowerRing` would be in an `item` package, with a class name of `ItemPowerRing`.
-* A `Block` called `NotDirt` would be in a `block` package, with a class name of `BlockNotDirt`.
-* Finally, a `TileEntity` for a block called `SuperChewer` would be a `tile` or `tileentity` package, with a class name of `TileSuperChewer`.
-
-Prepending your class names with what *kind* of object they are makes it easier to figure out what a class is, or guess the class for an object.
+Naming your classes, and being consistant, is important to knowing what those classes do. `BlockSauce` should be a Block, `ItemSpoon` should be an Item, et cetera. However, making sure those classes are in the right package will also help with the ability to find that class if you decide you need to look at it one day down the line. If you have `EntitySauceMonster` in the `com.examplemod.awesomesauce.client.gui` package, you might need to think about refactoring your code to better structure it.
```
Comment (LexManos) [inline docs/gettingstarted/structuring.md]: You need to make note that these are purely suggestions and not real good ones at that. Name things whatever ytou want as long as they start with Upper case and are descriptive enough to tell you what you want to know about the class.

**Comment by HACKhalo2** [comment]
Thank you for the suggestions guys, and @LexManos I changed the areas you pointed out.

File: docs/gettingstarted/structuring.md (hunk 7 above)
Comment (tterrag1098) [inline docs/gettingstarted/structuring.md]: This needs to go before merge

File: docs/gettingstarted/structuring.md [hunk 10]
```diff
@@ -1,54 +1,79 @@
...
+ * **ONCE YOU PICK THIS IDENTIFIER, YOU SHOULD NOT CHANGE IT DURING THE LIFE OF THE MOD. BAD THINGS HAPPEN IF YOU DO AND OTHER MODS DEPEND ON YOUR MOD.**
+* `name` is the *external* name of the mod, and takes a `String`. This is literally just the name of your mod, in our case it would be `name = "AwesomeSauce"`.
+* `version` is the version of the mod, and takes a `String`. There is no set standard for how you version things, but it's recommended that you read the [versioning](https://mcforge.readthedocs.io/en/latest/conventions/versioning/) article.
+
+Simple, right? `@Mod` has a wide range of other things you can throw at it, like the `modLanguage` example above, but those will be covered in the articles that use them.
```
Comment (condensedapple) [inline docs/gettingstarted/structuring.md]: I suggest to replace `modLanguage` with `modLanguageAdapter`, as it probably won't exist anymore in near future (see [here](https://github.com/MinecraftForge/MinecraftForge/pull/4029/files?diff=unified#diff-c8a307019565332befe89f9f61b2975bL161)).

**Review by elifoster — CHANGES_REQUESTED** [review]
